robot/rover/internal_comms/src/Serial.cpp

```diff
@@ -16,22 +17,42 @@ namespace internal_comms
 
     void readCommand(CommandCenter* commandCenter)
     {
-        Command* command = commandCenter->processCommand();
+        Command* command = CommandCenter::processCommand();
         if(command->isValid)
         {
             commandCenter->executeCommand(command->commandID, command->rawArgs, command->rawArgsLength);
         }
         else
         {
             // Handle invalid command
+            // (Create error message and put it at the front of the queue so that the 
+            // TX2 can resend)
         }
+
+        free(command->rawArgs);
+        command->rawArgs = nullptr;
+
+        delete command;
+        command = nullptr;
     }
 
-    void sendCommand(Command* command)
-    {
-        Serial.write(command->commandID);
-        Serial.write(command->rawArgsLength);
-        Serial.write(command->rawArgs, command->rawArgsLength);
+    void sendMessage(CommandCenter* commandCenter, int enablePin) {
+        if (digitalRead(enablePin)) {
```

Comment:
so what happens if digitalRead(enablePin) is false(?) shouldnt it queue the message?